duckduckgo/iOS
# Add initial NetP controller and provider classes

Task/Issue URL: https://app.asana.com/0/0/1205043071738197/f
BSK PR: https://github.com/duckduckgo/BrowserServicesKit/pull/418

**Description**:

Adds the initial classes needed for the `PacketTunnelProvider` extension and for controlling it from the iOS app along with some configuration. This PR also brings in the NetP iOS compatibility changes needed after some incompatibility was merged.

Note that until we’re ready to release, the extension will remain not embedded in the app and any Network Extension code will be compiler flagged



**Steps to test this PR**:
1. Check out this branch
2. Embed the `PacketTunnelProvider` extension into the app.
3. At this stage, just making sure everything compiles. There is no user facing change.

<!—
Before submitting a PR, please ensure you have tested the combinations you expect the reviewer to test, then delete configurations you *know* do not need explicit testing.

Using a simulator where a physical device is unavailable is acceptable.
—>

**Copy Testing**:


**Orientation Testing**:


**Device Testing**:


**OS Testing**:


**Theme Testing**:


—
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
